graspologic/pipeline/embed/laplacian_spectral_embedding.py

```diff
@@ -0,0 +1,247 @@
+    check_argument(dimensions >= 1, "dimensions must be positive")
+
+    check_optional_argument_types(elbow_cut, int, "elbow_cut must be an int or None")
+    check_argument(elbow_cut is None or elbow_cut >= 1, "elbow_cut must be positive")
+
+    check_argument_types(
+        svd_solver_algorithm, str, "svd_solver_algorithm must be a str"
+    )
+    check_argument(
+        svd_solver_algorithm in __SVD_SOLVER_TYPES,
+        f"svd_solver_algorithm must be one of the values in {','.join(__SVD_SOLVER_TYPES)}",
+    )
+
+    check_argument_types(
+        svd_solver_iterations, int, "svd_solver_iterations must be an int"
+    )
+    check_argument(svd_solver_iterations >= 1, "svd_solver_iterations must be positive")
+
+    check_optional_argument_types(svd_seed, int, "svd_seed must be an int or None")
+    check_argument(
+        svd_seed is None or 0 <= svd_seed <= 2 ** 32 - 1,
+        "svd_seed must be a nonnegative, 32-bit integer",
+    )
+
+    check_optional_argument_types(
+        regularizer,
+        numbers.Real,
+        "regularizer must be of type int, float, or some subclass of numbers.Real",
+    )
+    check_argument(
+        regularizer is None or regularizer >= 0, "regularizer must be nonnegative"
+    )
+
+    check_argument_types(
+        graph,
+        (nx.Graph, nx.DiGraph, nx.OrderedGraph, nx.OrderedDiGraph),
+        "graph must be of type networkx.Graph, networkx.DiGraph, "
+        "networkx.OrderedGraph, networkx.OrderedDiGraph",
+    )
+    check_argument(
+        not graph.is_multigraph(),
+        "Multigraphs are not supported; you must determine how to represent at most "
+        "one edge between any two nodes, and handle the corresponding weights "
+        "accordingly",
+    )
+
+    if not is_real_weighted(graph, weight_attribute=weight_attribute):
+        warnings.warn(
+            f"Graphs with edges that do not have a real numeric weight set for every "
+            f"{weight_attribute} attribute on every edge are treated as an unweighted "
+            f"graph - which presumes all weights are `1.0`. If this is incorrect, "
+            f"please add a '{weight_attribute}' attribute to every edge with a real, "
+            f"numeric value (e.g. an integer or a float) and call this function again."
+        )
+        weight_attribute = None  # this supercedes what the user said, because
+        # not all of the weights are real numbers, if they exist at all
+        # this weight=1.0 treatment actually happens in nx.to_scipy_sparse_matrix()
+
+    graph_as_csr = nx.to_scipy_sparse_matrix(graph, weight=weight_attribute)
+
+    if not is_fully_connected(graph):
+        warnings.warn("More than one connected component detected")
+
+    graph_as_csr = remove_loops(graph_as_csr)
+
+    graph_as_csr = pass_to_ranks(graph_as_csr)
+
+    node_labels = np.array(list(graph.nodes()))
+
+    embedder = LaplacianSpectralEmbed(
+        form="R-DAD",
```

Comment:
just pointing out that 'form' is not passed down. though based on our discussion on teams I am feeling like that is a good thing and we can just take out the "form" argument

Reply:
oh crap that's a good catch - it absolutely should be passed down.  that was an oversight on my part.

but yeah, I hear you about the form stuff - DAD is just R-DAD where regularizer = 0.  But maybe someone will want to run some I-DAD experiments later or something, I don't know.